src/debonair/Order.java

```diff
@@ -0,0 +1,125 @@
+// user defined package
+package debonair;
+
+// importing built in packages
+import java.time.LocalDateTime;
+import java.util.*;
+
+// importing classes from other packages
+import resources.Constants;
+
+public class Order {
+	// class variables
+	private int orderId;
+	private LocalDateTime orderDate = LocalDateTime.now();
+	private Set<Item> orderCart;
+	private static Map<Integer, Bill> currentSale = new HashMap<>();
+
+	// Getter-Setters
+	public int getOrderId() {
+		return orderId;
+	}
+
+	public void setOrderId(int orderId) {
+		this.orderId = orderId;
+	}
+
+	public LocalDateTime getOrderDate() {
+		return orderDate;
+	}
+
+	public void setOrderDate(LocalDateTime orderDate) {
+		this.orderDate = orderDate;
+	}
+
+	public Set<Item> getOrderCart() {
+		return orderCart;
+	}
+
+	public void setOrderCart(TreeSet<Item> orderCart) {
+		this.orderCart = orderCart;
+	}
+
+	public static Map<Integer, Bill> getCurrentSale() {
+		return currentSale;
+	}
+
+	public static void setCurrentSale(HashMap<Integer, Bill> currentSale) {
+		Order.currentSale = currentSale;
+	}
+
+	// Overridden toString method
+	@Override
+	public String toString() {
+		return "Order [orderId=" + orderId + ", orderDate=" + orderDate + ", orderCart=" + orderCart + "]";
+	}
+
+	// Parameterized constructor
+	public Order(int orderId, LocalDateTime orderDate, Set<Item> orderCart) {
+		super();
+		this.orderId = orderId;
+		this.orderDate = orderDate;
+		this.orderCart = orderCart;
+	}
+
+	/**
+	 * Function to create order of the customer
+	 */
+	public static void createOrder() {
+		Random randomObj = new Random();
+		// Generating Random Order Id
+		int orderId = randomObj.nextInt(100);
+		// Getting current Date Time
+		LocalDateTime currentDateTime = LocalDateTime.now();
+		@SuppressWarnings("resource")
+		Scanner sc = new Scanner(System.in);
+		String choice;
+		TreeSet<Item> itemSet = new TreeSet<>();
+		// User adding items to the cart
+		do {
+			System.out.println("Select item Id you want to add to the cart");
+			int itemNumber = sc.nextInt();
+			if (itemSet.add(Menu.ITEM_MENU.get(itemNumber))) {
+				System.out.println("The added item is: " + Menu.ITEM_MENU.get(itemNumber).getItemName());
+			} else {
+				System.err.println("Failed to add the item");
+			}
```

Comment:
Here two situations arise code will go to else if user entered an item no that does not exist or that item was already present in the set.
Handle case no 1 separately and ask user if he wants to update quantity for the already added item.
Handle case 2 and inform user that item was not found

Reply:
This is not a part of the assignment but will help you to judge corner scenarios and help u practice more with collections